### PR TITLE
EN-1838 [Node.js SDK] Enhanced custom logger

### PR DESCRIPTION
**Pull request by vladimir-grezin**
Pull Request Checklist
======================

Please read this for our PR Culture:
------------------------------------
https://incountry.atlassian.net/wiki/spaces/ED/pages/46858869/Pull+requests+culture+Draft

Work relates to:
----------------
Link(s) to Jira Task/User Story/Bug/Epic:
- https://incountry.atlassian.net/browse/EN-1838

Code Quality (Before submitting PR)
------------
- [x] Ensure code compiles and passes Sonar Lint.
- [x] Do your best to ensure integration tests pass. If they don't then...
- [x] Remove, ignore, or mark pending any brittle tests.
- [x] Remove commented out code.

Target Release
--------------
- [ ] Please put the target release vehicle this needs to be bundled with.
- [ ] Did you increment the version in package.json? What is the new version?

### CODE REVIEW

File: storage.js [hunk 1]
```diff
@@ -6,30 +6,56 @@ const defaultLogger = require('./logger');
 const CountriesCache = require('./countries-cache');
 const SecretKeyAccessor = require('./secret-key-accessor');
 const { InCrypt } = require('./in-crypt');
+const { isPositiveInt, parsePoPError } = require('./utils');
+const { StorageServerError } = require('./errors');
+
+/**
+ * @typedef Record
+ * @property {string} key
+ */
```
Comment (RommeroInCountry) [inline storage.js]: Merge error?

Comment (vladimir-grezin) [reply storage.js]: Nope. This typedef is used bellow

File: tests/unit/storage-tests.js [hunk 2]
```diff
@@ -41,11 +41,13 @@ const TEST_RECORDS = [
 ];
 
 describe('Storage', function () {
+  this.timeout(5000);
```
Comment (igor-alekseev-dev) [inline tests/unit/storage-tests.js]: Why did you decide to increase the timeout?

Comment (vladimir-grezin) [reply tests/unit/storage-tests.js]: No need for it. Removed.

File: storage.js [hunk 3]
```diff
@@ -295,70 +452,98 @@ class Storage {
...
+      }, requestOptions);
     }
+
     throw new Error('Record not found');
   }
 
-  async deleteAsync(request) {
+  async deleteAsync(record, requestOptions = {}) {
+    let endpoint;
+    let key;
```
Comment (artemgolozubov) [inline storage.js]: Looks like endpoint and key could be constants.

**Pull request merged**